### PR TITLE
Feat/add background geolocation

### PR DESCRIPTION
**Pull request by zatteo**
Full port of CozyGPSMemory. Allow to manage from a front end geolocation tracking.

No submodule at the end of the PR. cozy-flagship-app will be added as a submodule on CozyGPSMemory (already working locally and PR coming soon).

Also included : 
- `isAvailable` generic method
- `checkPermissions` and `requestPermission` generic method

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -251,6 +255,11 @@ dependencies {
             }
         }
     }
+    implementation ("androidx.appcompat:appcompat:1.3.1") {
+        version {
+            strictly '1.3.1'
+        }
+    }
```
Comment (Crash--) [inline android/app/build.gradle:262]: c'est marrant qu'ici tu mettes 1.3.1, mais que plus bas ce soit 1.4.2. https://github.com/cozy/cozy-flagship-app/pull/935/files#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65aR11

File: ios/CozyReactNative/Info.plist [hunk 2]
```diff
@@ -118,8 +122,14 @@
 	<string>Afin de prendre des photos pour Drive ou MesPapiers afin d&apos;importer des documents</string>
 	<key>NSFaceIDUsageDescription</key>
 	<string>Permet le déverouillage de l&apos;application en utilisant FaceID</string>
+	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
+	<string>Location required in background</string>
```
Comment (Crash--) [inline ios/CozyReactNative/Info.plist:126]: faudra mettre les bons textes. Tu les as ?

File: ios/CozyReactNative/Info.plist [hunk 3]
```diff
@@ -118,8 +122,14 @@
...
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string></string>
+	<string>Location required when app is in use</string>
+	<key>NSMotionUsageDescription</key>
+	<string>Motion permission helps detect when device in in-motion</string>
+	<key>NSPhotoLibraryUsageDescription</key>
+	<string>Pour faire un backup de vos photos</string>
```
Comment (Crash--) [inline ios/CozyReactNative/Info.plist:132]: a retirer de cette PR?

File: ios/CozyReactNative/Info.plist (hunk 2 above)
Comment (zatteo) [reply ios/CozyReactNative/Info.plist:117]: Je vais les demander

File: android/app/build.gradle (hunk 1 above)
Comment (zatteo) [reply android/app/build.gradle:266]: Ça avait pas de sens en effet... J'ai enlevé la mention du 1.4.2

**Review by Ldoppea — APPROVED** [review]

**Comment by Ldoppea** [comment]
It is difficult to tell if every geolocation high-level methods are protected by try...catch bocks. Did you check that it is the case?

File: ios/CozyReactNative/Info.plist [hunk 4]
```diff
@@ -106,11 +110,17 @@
 		</dict>
 	</dict>
 	<key>NSCameraUsageDescription</key>
-	<string>Afin de prendre des photos pour Drive ou MesPapiers afin d'importer des documents</string>
+	<string>Afin de prendre des photos pour Drive ou MesPapiers afin d&apos;importer des documents</string>
```
[the remark this answers is not in the document]
Comment (zatteo) [reply ios/CozyReactNative/Info.plist]: Done

File: src/app/domain/geolocation/services/permissions.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        granted:
+          fineLocationPermission.granted &&
+          activityRecognitionPermission.granted,
+        canRequest:
+          fineLocationPermission.canRequest &&
+          activityRecognitionPermission.canRequest
+      }
+    }
+
+    throw new Error(t('services.backup.errors.platformNotSupported'))
```
[the remark this answers is not in the document]
Comment (zatteo) [reply src/app/domain/geolocation/services/permissions.ts:46]: Nice catch! I added a new translation more generic

**Pull request merged**